share/eds/eds_test.go

```diff
@@ -0,0 +1,121 @@
+package eds
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	ds "github.com/ipfs/go-datastore"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	carv1 "github.com/ipld/go-car"
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/ipld"
+)
+
+func TestQuadrantOrder(t *testing.T) {
+	// TODO: add more test cases
+	nID := []byte{0, 0, 0, 0, 0, 0, 0, 0}
+	parity := append(appconsts.ParitySharesNamespaceID, nID...) //nolint
```

Comment:
what's nolint for?

Reply:
appendAssign: append result not assigned to the same slice (gocritic)

Reply:
ack